### ISSUE TITLE
Unable to map $GOPATH when debugging remotely

### ISSUE THREAD

**Issue by hgl** [microsoft/vscode-go#3177]
**Is your feature request related to a problem? Please describe.**
When debugging a go program living inside a docker container, I'm able to map my own source code with `remotePath` and `cwd`, but I have no way to map `$GOPATH`. In docker it starts with `/go`, but on my host, not the case. So whenever I try to descend into a 3rd party package, I'm prompted with a file not found error.

**Describe the solution you'd like**
Maybe another option in launch.json to map `$GOPATH`? (Or is there already one that I missed?)

Since vscode should be able infer `$GOPATH` on the host, this option should only refer to remote `$GOPATH` imho.

**Comment by hyangah** [comment]
@quoctruong @ramya-rao-a 
I see only [a special handling of `GOROOT` mapping](https://github.com/microsoft/vscode-go/blob/master/src/debugAdapter/goDebug.ts#L841-L858) for https://github.com/Microsoft/vscode-go/issues/1178 but no `GOPATH` mapping for remote debugging. Am I missing something?

**Comment by hyangah** [comment]
Moreover, the `GOROOT` handling logic blindly replaces the part before `/src/` with `GOROOT`, so it will interfere the correct mapping of source code under `$GOPATH/src`.
If the headless `dlv` does not provide info about its runtime environment, I think we need explicit mappings.

**Comment by hyangah** [comment]
It seems the issue https://github.com/microsoft/vscode-go/pull/3079 is trying to fix is also related.
@fnmunhoz

**Comment by quoctruong** [comment]
This is now fixed in the nightly Go build (https://github.com/golang/vscode-go). @hgl please give it a try: https://github.com/golang/vscode-go/blob/master/docs/nightly.md

**Comment by hgl** [comment]
@quoctruong Thanks. Will do when time is available. How do I specify the remote $GOPATH? (Or does delve report the remote GOPATH value so there is nothing to be configured?)

**Comment by stamblerre** [comment]
@hgl: Based on https://github.com/microsoft/vscode-go/issues/3241, it sounds like you will not have to configure the remote GOPATH.

**Comment by hgl** [comment]
This looks pretty awesome. So even `remotePath` can be omitted now?

**Comment by stamblerre** [comment]
Sounds like it, but I'll let @quoctruong confirm.

**Comment by quoctruong** [comment]
@hgl yes that’s correct. The idea is that you won’t have to specify it.

**Comment by hgl** [comment]
@quoctruong I tried to test It just now, but remote debugging seems to be broken in nightly. 

When I clicked the green triangle to start debugging in the debug panel, the debug session would automatically close after a couple of seconds, and there is no message in debug console or output, and the `dlv` output kept showing `API server listening at: [::]:2000`, without changing to my binary's normal output, which means it had never connected.

I could only connect after switching to the stable version of the extension.

My launch setting:

```
{
    "version": "0.2.0",
    "configurations": [
        {
        "name": "debug go",
        "type": "go",
        "request": "attach",
        "mode": "remote",
        "host": "127.0.0.1",
        "port": 2000
      }
    ]
}
```
